daemon/api_sideload_n_try.go

```diff
@@ -299,13 +362,21 @@ func sideloadSnap(st *state.State, snapFile *uploadedSnap, flags sideloadFlags)
 		instanceName = sideInfo.RealName
 	}
 
-	tset, _, err := snapstateInstallPath(st, sideInfo, snapFile.tmpPath, instanceName, "", flags.Flags, nil)
+	var tset *state.TaskSet
+	container := "snap"
+	if compInfo == nil {
+		tset, _, err = snapstateInstallPath(st, sideInfo, snapFile.tmpPath, instanceName, "", flags.Flags, nil)
+	} else {
+		// It is a component
+		container = "component"
+		tset, err = snapstateInstallComponentPath(st, snap.NewComponentSideInfo(compInfo.Component, snap.Revision{}), snapInfo, snapFile.tmpPath, flags.Flags)
+	}
 	if err != nil {
 		return nil, errToResponse(err, []string{sideInfo.RealName}, InternalError, "cannot install snap file: %v")
 	}
 
-	msg := fmt.Sprintf(i18n.G("Install %q snap from file %q"), instanceName, snapFile.filename)
-	chg := newChange(st, "install-snap", msg, []*state.TaskSet{tset}, []string{instanceName})
+	msg := fmt.Sprintf(i18n.G("Install %q %s from file %q"), instanceName, container, snapFile.filename)
+	chg := newChange(st, "install-"+container, msg, []*state.TaskSet{tset}, []string{instanceName})
 	chg.Set("api-data", map[string]string{"snap-name": instanceName})
 
```

Comment:
we should leave a TODO in try code as well about supporting components?

Reply:
TODO added